### ISSUE TITLE
Backslash + quote mark causes Doxygen not to generate documentation

### ISSUE THREAD

**Issue by nilsbrynedal** [doxygen/doxygen#10586]
In C#, the presence of **\\"** (a backslash followed by a quote mark) causes Doxygen to not generate documentation for a class:

```
namespace TestNameSpace
{
    public static class ExampleClass
    {
        public static string ExampleProperty => "Test\"; // remove the backslash and the bug will not appear
    }
}

```
This will cause Doxygen to only create a page for the ExampleClass.cs file, but no information of the class itself.
If one removes the **\\** and re-runs Doxygen, then the generated documentation for ExampleClass will contain much more information.

I suspect that the backslash is seen as an escape character, which then causes the rest of the text to be incorrect (wrong number of quotes?). But this is in the code itself, not in the comments, so Doxygen should not care about this.

Is there any quick way of getting around this bug? Can I add something that prevents this backslash from being handled as an escape character?
I cannot change the source code to prevent this bug.

`doxygen -v` gives this output:
**1.10.0 (ebc57c6dd303a980bd19dd74b8b61c8f3f5180ca)**

All files needed to replicate this error is found in:
[Doxygen error.zip](https://github.com/doxygen/doxygen/files/14022996/Doxygen.error.zip)

If anything is unclear or if more info would be needed, let me know and I'll try to answer as good as I can.

**Comment by albert-github** [comment]
Doxygen sees indeed the `\"` as an escaped `"` and thus the string as not terminated.
I'm amazed that this compiles with C#.
- Where in the C# standard is it defined that a `\` (backslash) in a string doesn't have a special meaning?
- What "escape sequences" are escape sequences and what ain't?

**Comment by nilsbrynedal** [comment]
Thanks for such a quick response. Let me give some more background.
First, I realized that I didn't tell my OS, I'm running on Windows 10 Enterprise.

I can't provide the full source code, but the class that caused me to find this bug contains two properties:
```
public static string? MainDirectory { get; private set; }
public static string LogData => MainDirectory + @"\Log\";
```
These relate to directories. Other variables will append more stuff after `LogData` to make a full path.
As you see, the backslash is inside quotes so there will be no syntax issues preventing compilation.

**Comment by albert-github** [comment]
There is one essential information in the comment https://github.com/doxygen/doxygen/issues/10586#issuecomment-1906243406 as here the `@` string syntax is used! and I don't see the `@` in the original post.
Doxygen does try to handle the `@` but maybe a case with `\"` escaped the attention (especially at the end of a string).

- Is in your original code the `@` present (and missed by the copy / paste / typing of the issue)?

There is no answer to the questions I asked before.

**Comment by nilsbrynedal** [comment]
Hi
I made the reproducer code as simple as possible, hence the absence of the `@`. I added the `@` to the example code (so `@"Test\"` instead of `"Test\"`) and the problem was the same. I've attached the new code.
[Doxygen error version 2.zip](https://github.com/doxygen/doxygen/files/14035106/Doxygen.error.version.2.zip)

I'm afraid I can't give a full answer to your question regarding the C# standard or escape sequences. However, the backslash is indeed an escape character in C#. E.g. if you write a string `"\""` it will be seen as a string containing a `"` character.
The `@` sign before a string means that the string is to be interpreted literally, so `@"\"" `would result in the string `\"`.

I hope this answers your questions.

I tried a third way. This time by removing the `@` and instead having double backslashes. So `"Test\\"` instead of `"Test\"`.
This give me proper Doxygen output. **Could it be that Doxygen handles double backslashes fine but does not interpret the `@` before a string as it should?**

Here is the code for that:
[Doxygen error version 3, double backslash.zip](https://github.com/doxygen/doxygen/files/14035313/Doxygen.error.version.3.double.backslash.zip)

What surprises me is that Doxygen seems to care _at all_ of the source code itself, I thought Doxygen only read the tripple slash comments (those before a class/method)?
But I guess that it has to go through _all_ characters to see what is a comment and what isn't?

**Comment by albert-github** [comment]
So the original example was incorrect source code.

I already read up the C# strings and I see that the implementation of the `@"` is not 100% correct, this has to be adjusted.

Doxygen does need to read to read all teh code, not only to find where the comment signs are (e.g. it is possible to have tripple slashes in strings and this is not comment), needs it for code display (`SOURCE_BROWSER`), for call / caller graphs, for relation between classes etc. etc.
Doxygen does not support only `///` comments but also `/**`, `//!`, `/*!`. `//!<`, `/*!<` etc. (see the manual).

**Comment by nilsbrynedal** [comment]
Indeed, I didn't realize the source code was that important so I didn't double check the syntax. Sorry about that.

Thank you for your help. Do you have any notion of how soon this bug would be fixed? Days, months, years?

**Comment by albert-github** [comment]
> Thank you for your help. Do you have any notion of how soon this bug would be fixed? Days, months, years?

Hopefully I can create a proposed patch in the next few days.

**Comment by nilsbrynedal** [comment]
Thank you Albert, keep up the good work! Doxygen is great.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #10591

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
